### ISSUE TITLE
Date type takes up a lot of vertical space in grid mode print view

### ISSUE THREAD

**Issue by lognaturel** [enketo/enketo-express#141]
In the online grid view, questions of type date take up more vertical space than expected when compared to other field types but it's very usable:
<img width="232" alt="Screen Shot 2020-04-10 at 3 53 33 PM" src="https://user-images.githubusercontent.com/967540/79028040-74279d80-7b43-11ea-97b4-3d69684e3c82.png">

In print view they take up too much vertical space to be usable for printing forms to be filled out:
<img width="68" alt="Screen Shot 2020-04-10 at 3 55 16 PM" src="https://user-images.githubusercontent.com/967540/79028111-b7820c00-7b43-11ea-9785-55fb6473ccfe.png">

**Comment by lognaturel** [comment]
I'm wondering whether there might be deeper issues with the grid theme print view. Below is some of what I see from https://enke.to/grid with  
<img width="479" alt="Screen Shot 2020-04-10 at 4 10 17 PM" src="https://user-images.githubusercontent.com/967540/79028651-c5d12780-7b45-11ea-91a2-2fe30bc5f5dc.png">

Google Chrome | 80.0.3987.163 (Official Build) (64-bit)
-- | --
Revision | e7fbe071abe9328cdce4ffedac9822435fbd3656-refs/branch-heads/3987@{#1037}
OS | macOS Version 10.15.2 (Build 19C57)
JavaScript | V8 8.0.426.30

**Comment by MartijnR** [comment]
Thank you! Are you seeing this when using the Print button on top of the form?

There is lot of javascript magic going on to print Grid theme forms in the same way as they look on the screen, because for some reason multiline flexboxes do not work in print views (in none of the browsers afaik). Hence, the code is actually reverse engineering a CSS engine... 

Side note:
The other issue with printing is browser differences, hence we now have the PDF endpoints (that use a headless Chrome browser). And if printing is really important (as it is for e.g. OpenClinica and perhaps the user you are working with), it may make a lot of sense to consume one of these API endpoints instead.

**Comment by lognaturel** [comment]
> Are you seeing this when using the Print button on top of the form?

Ah! I had seen slightly different behavior at different times but I hadn't noticed the pattern. There is indeed one slight difference that I can see between using the print button and not: the grid lines aren't broken as in my comment's screenshot. However, the lines for text are at inconsistent vertical locations and the date is still too tall:
<img width="470" alt="Screen Shot 2020-04-13 at 1 30 18 PM" src="https://user-images.githubusercontent.com/967540/79158386-19808280-7d8b-11ea-9583-731fdbc545bf.png">

> multiline flexboxes do not work in print views

That's really annoying, sorry.

> the PDF endpoints (that use a headless Chrome browser)

That sounds like it would be useful. I'm using Chrome, though, so it seems that I'd get a similar result? I've tried it but am getting "Survey information not complete or invalid" which is confusing to me because I can successfully use the same auth, `server_url`, `form_id` combo for other endpoints.

**Comment by MartijnR** [comment]
Thank you. We'll leave this issue to push down those dotted lines. 

Yes, it would be similar for Chrome users to use the print button or pdf endpoint.

>  "Survey information not complete or invalid" which is confusing to me

That is puzzling indeed. Will look into this asap!

**Comment by lognaturel** [comment]
> That is puzzling indeed. Will look into this asap!

I'm sorry, this was me not being familiar with the API! I thought that if I had already gotten a survey URL I could use the `GET` variant for the PDF but no, it looks like I have to use `POST`.

**Comment by MartijnR** [comment]
No, you were right. If an Enketo ID has already been generated, a GET should work, so I think that may be a bug!
